File: Programming space/Java Programs/NonDigitMaxNum.java
public class NonDigitMaxNum{

    public static int maxOfnonDigit(int number,int nonDigit){
        
        for(int i=number; i>0; i--){
            if(!String.valueOf(number).contains(String.valueOf(nonDigit)))
                return number;
            else
                number--;
        }
        return 0;
    }
    public static void main(String[] args){
        int number = 1000;
        int nonDigit = 9;
        int result =  maxOfnonDigit(number,nonDigit);
        System.out.println("The maximum number is "+result);
    }
}

File: Programming space/Pyhton Programs/NonDigitMaxNum.py
def max_of_non_digit(number,non_digit):
    while(number > 0):
        if str(non_digit) not in str(number):
            return number
        else:
            number -= 1
    return 0;

if __name__ == "__main__":
    number = 143
    non_digit = 4
    result = max_of_non_digit(number,non_digit)
    print(result)
